pyro/distributions/torch_distribution.py

```diff
@@ -15,7 +16,7 @@
 from .util import broadcast_shape, scale_and_mask
 
 
-class TorchDistributionMixin(Distribution):
+class TorchDistributionMixin(Distribution, Callable):
```

Comment:
This is a hack so that mypy understands `assert isinstance(msg["fn"], TorchDistributionMixin)`

Reply:
Using `@Callable.register` doesn't seem to work (the code after `assert isinstance(msg["fn"], TorchDistributionMixin)` is unreachable). Maybe this has to do with `typing.Callable` vs `collections.abc.Callable`?

It is strange that `assert isinstance(msg["fn"], TorchDistributionMixin)` doesn't work in the first place because `assert isinstance(msg["fn"], Distribution)` works fine. Seems like a Mypy bug.

In the following PR I want to add `warn_unreachable=True` mypy flag that can catch cases like this, otherwise unreachable code is not checked by mypy and fails silently.

Reply:
Well thanks for checking!